prowler/providers/aws/aws_provider.py

```diff
@@ -881,7 +886,134 @@ def get_checks_to_execute_by_audit_resources(self) -> set[str]:
+            session = (
+                session if session else AwsProvider.setup_session(mfa_enabled, profile)
+            )
+
+            # Test Connection using the IAM Role
+            if role_arn:
+                session_duration = validate_session_duration(session_duration)
+                role_session_name = validate_role_session_name(role_session_name)
+                role_arn = parse_iam_credentials_arn(role_arn)
+                assumed_role_information = AWSAssumeRoleInfo(
+                    role_arn=role_arn,
+                    session_duration=session_duration,
+                    external_id=external_id,
+                    mfa_enabled=mfa_enabled,
+                    role_session_name=role_session_name,
+                )
+                assumed_role_credentials = AwsProvider.assume_role(
+                    session,
+                    assumed_role_information,
+                )
+                session = Session(
+                    aws_access_key_id=assumed_role_credentials.aws_access_key_id,
+                    aws_secret_access_key=assumed_role_credentials.aws_secret_access_key,
+                    aws_session_token=assumed_role_credentials.aws_session_token,
+                    region_name=aws_region,
+                    profile_name=profile,
+                )
+
+            sts_client = AwsProvider.create_sts_session(session, aws_region)
+            caller_identity = sts_client.get_caller_identity()
+            # Include the region where the caller_identity has validated the credentials
+            return AWSCallerIdentity(
+                user_id=caller_identity.get("UserId"),
+                account=caller_identity.get("Account"),
+                arn=ARN(caller_identity.get("Arn")),
+                region=aws_region,
+            )
+
+        except (ClientError, ProfileNotFound, ArgumentTypeError) as error:
+            logger.error(
+                f"{error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
+            )
+            raise error
+
+        except Exception as error:
+            logger.critical(
+                f"{error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
+            )
+            raise error
+
+    @staticmethod
+    def create_sts_session(
+        session: Session, aws_region: str = AWS_STS_GLOBAL_ENDPOINT_REGION
+    ) -> Session.client:
+        """
+        Create an STS session client.
+
+        Parameters:
+        - session (session.Session): The AWS session object.
+        - aws_region (str): The AWS region to use for the session.
+
+        Returns:
+        - session.Session.client: The STS session client.
+
+        Example:
+            session = boto3.session.Session()
+            sts_client = create_sts_session(session, 'us-west-2')
+        """
+        try:
+            sts_endpoint_url = (
+                f"https://sts.{aws_region}.amazonaws.com"
+                if "cn-" not in aws_region
```

Comment:
This could potentially lead to unwanted behaviors, unless you are 100% certain about it. Consider using compiled regexes, `.startswith` (if that is a prefix) or something similar. Just a suggestion, not a requested change.

Reply:
I think I highlighted too many lines. I was referring to:

```python
if "cn-" not in aws_region
```

Reply:
Great catch 👏